### ISSUE TITLE
Installing the front-end requires node <= v6.9.0

### ISSUE THREAD

**Issue by falquaddoomi** [BRCAChallenge/brca-exchange#899]
Attempting to install the front-end dependencies from our current package.json with a post-6.9.0 version of node.js fails with node-gyp compiler errors. The errors are introduced by dependencies of the custom version of `muts-needle-plot` that we're using, specifically this dependency chain: `prepare@0.0.6 -> jsdom@1.5.0 -> contextify@0.1.15`.

The complete log is below:

```
$ npm install
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@3.0.11: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated swig@1.4.2: This package is no longer maintained

> fibers@3.0.0 install /Users/Faisal/Checkouts/brcaex-test/website/node_modules/fibers
> node build.js || nodejs build.js

`darwin-x64-57` exists; testing
Binary is fine; exiting

> contextify@0.1.15 install /Users/Faisal/Checkouts/brcaex-test/website/node_modules/contextify
> node-gyp rebuild

make: Entering directory '/Users/Faisal/Checkouts/brcaex-test/website/node_modules/contextify/build'
  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc:131:56: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                       ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8.h:3901:3: note: 'NewInstance' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version", Local<Object> NewInstance() const);
  ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:150:16: error: no member named 'SetAccessCheckCallbacks' in 'v8::ObjectTemplate'
        otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
        ~~~~~  ^
../src/contextify.cc:182:51: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                  ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8.h:3453:3: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version",
  ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:209:38: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                     ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8.h:3453:3: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version",
  ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
../src/contextify.cc:210:42: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
            !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                         ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8.h:3453:3: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version",
  ^
/Users/Faisal/.node-gyp/8.12.0/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))
                            ^
4 warnings and 1 error generated.
make: *** [contextify.target.mk:125: Release/obj.target/contextify/src/contextify.o] Error 1
make: Leaving directory '/Users/Faisal/Checkouts/brcaex-test/website/node_modules/contextify/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/Faisal/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Darwin 17.7.0
gyp ERR! command "/Users/Faisal/.nvm/versions/node/v8.12.0/bin/node" "/Users/Faisal/.nvm/versions/node/v8.12.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/Faisal/Checkouts/brcaex-test/website/node_modules/contextify
gyp ERR! node -v v8.12.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN expose-loader@0.7.5 requires a peer of webpack@^2.0.0 || ^3.0.0 || ^4.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN brcaexchange@3.54.0-alpha.0 No repository field.
npm WARN brcaexchange@3.54.0-alpha.0 No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the contextify@0.1.15 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/Faisal/.npm/_logs/2018-09-25T18_39_22_335Z-debug.log
```

**Comment by zfisch** [comment]
ah yes... this one has been haunting us for a while. it would be great to sort it out so we can upgrade node 👍

**Comment by falquaddoomi** [comment]
I don't appear to have write access to [BRCAChallenge/muts-needle-plot](https://github.com/BRCAChallenge/muts-needle-plot), so I've created another fork of [cmarkello/muts-needle-plot](https://github.com/cmarkello/muts-needle-plot) (the repo referenced in the site's package.json) at [falquaddoomi/muts-needle-plot](https://github.com/falquaddoomi/muts-needle-plot). The changes are in the `polish_sandbox` branch, since that's the branch from cmarkello's repo that we were referencing.

Specifically, my fork removes `prepare` (since it was apparently never used in the project), and upgrades `d3` to 3.5.4, the closest version to the previously-installed 3.5.1 that doesn't have a dependency on `jsdom` (and thus `contextify`). I also upgraded `watchify` since the older version fails to build on newer versions of node for different reasons (I can provide a log if need be; it's basically just a bunch of missing symbols that the older version of `fsevent` on which it depends expects to exist.)

After changing `muts-needle-plot` to my fork, the website dependencies install fine on node v8.12.0, the latest stable release. The lollipop charts also appear to work correctly: both BRCA1/2 Pathogenicity and Allele Frequency subviews render fine, and they support the view selection, exon zoom, and variant linking actions.

Anyway, I'm unsure if I should I should submit a pull request to `cmarkello/muts-needle-plot` or if I could perhaps get write access to `BRCAChallenge/muts-needle-plot` and move my changes there, then update the package.json dependency to the BRCAChallenge fork. I prefer the latter solution myself.
